trunk/lib/PostGalleryImageList.php

```diff
@@ -94,60 +100,112 @@ public static function get( int $postid = null ): array {
-                    }
-
-                    $alt = '';
-                    $imageTitle = '';
-                    $imageOptions = '';
-                    $imageDesc = '';
-                    $attachmentId = PostGalleryImage::checkForAttachmentData( $fullUrl, $postid );
-                    if ( !empty( $attachmentId ) ) {
-                        $attachment = get_post( $attachmentId );
-                        $alt = get_post_meta( $attachmentId, '_wp_attachment_image_alt', true );
-                        $imageOptions = get_post_meta( $attachmentId, 'postgallery-image-options', true );
-                        if ( !empty( $attachment ) ) {
-                            $imageTitle = $attachment->post_title;
-                            $imageDesc = $attachment->post_content;
-                        }
-                    }
-
-                    $imageOptionsParsed = PostGalleryImage::parseImageOptions( $imageOptions );
-
-                    $images[$file] = [
-                        'filename' => $file,
-                        'path' => $path,
-                        'url' => $fullUrl,
-                        'thumbURL' => get_bloginfo( 'wpurl' ) . '/?loadThumb&amp;path=' . $uploadUrl . '/' . $file,
-                        'title' => $imageTitle,
-                        'desc' => $imageDesc,
-                        'alt' => $alt,
-                        'post_id' => $postid,
-                        'post_title' => get_the_title( $postid ),
-                        'imageOptions' => $imageOptions,
-                        'imageOptionsParsed' => $imageOptionsParsed,
-                        'attachmentId' => $attachmentId,
-                        'srcset' => wp_get_attachment_image_srcset( $attachmentId, 'full' ),
-                        //'srcsetSizes' => wp_get_attachment_image_sizes($attachmentId, 'full'),
-                    ];
-                }
+        if ( !file_exists( $uploadDir ) || !is_dir( $uploadDir ) ) {
+            return [];
+        }
+        $dir = scandir( $uploadDir );
+
+        foreach ( $dir as $file ) {
+            if ( is_dir( $uploadDir . '/' . $file ) ) {
+                continue;
+            }
+            $fullUrl = $uploadFullUrl . '/' . $file;
+            $path = $uploadUrl . '/' . $file;
+
+            if ( PostGalleryImage::urlIsThumbnail( $fullUrl ) ) {
+                continue;
             }
+            $attachmentId = PostGalleryImage::checkForAttachmentData( $fullUrl, $postid );
+
+            $info = self::getAttachmentInfo( $attachmentId, $postid );
+            $file = $info['filename'];
+            $images[$file] = $info;
         }
 
         $images = self::sort( $images, $postid );
         self::$cachedImages[$postid] = $images;
         return $images;
     }
 
+    /**
+     * Get list with attachments-data from an attachment-list
+     *
+     * @param array $imageIdList
+     * @return array
+     */
+    public static function getByDynamic( $imageIdList ): array {
```

Comment:
Rethink namings.
What is a "Dynamic"? A `$imageIdList` is expected so `getByImageIdList` would be obvious.
The PhpDocs are telling something about attachment-data and attachments-list.
Please rename the everything according to what this method actualy does?

Reply:
Renamed to getByAttachmentIds( $attachmentIds )